webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/api/DataStructures.scala

```diff
@@ -58,53 +57,102 @@ final case class ApiAccountId(value: String) extends AnyVal
 final case class ApiAccountName(value: String) extends AnyVal
 
 /**
- * The actual authentication token.
+ * The actual authentication token, in clear text.
  *
- * TODO: Once support for plain text tokens is dropped, make separate types for plain and hashed tokens.
-         Current situation is confusing, and hence a bit risky.
+ * All tokens are 32 alphanumeric characters, optionally
+ * followed by a "-system" suffix, indicating a system token.
  *
- * There are two versions of tokens:
+ */
+final case class ApiTokenSecret(private val secret: String) {
+  // Avoid printing the value in logs, regardless of token type
+  override def toString: String = "[REDACTED ApiTokenSecret]"
+
+  // Prevent comparison
+  override def equals(obj: Any) = false
```

Comment:
this is confusing as since the Object equality and hashcode in Java follows some rules, here an object would not be equal to itself !

I think the good practice would be to avoid using the default `equals` method, and use another `equalsToken()`, or a `def isEqual(this: ApiTokenSecret, that: ApiTokenSecret)` instead. Many examples apply this, such as for date time types that are also subject to custom comparison logic.
When doing that we should document that the equals method should not be used, it all relies on code convention !